imports/plugins/core/checkout/client/templates/checkout/payment/methods/cards.js

```diff
@@ -42,3 +45,89 @@ Template.corePaymentMethods.helpers({
     return self;
   }
 });
+Template.payWithWallet.onCreated(function bodyOnCreated() {
+  this.state = new ReactiveDict();
+  this.state.set("details", { balance: 0 });
+  this.autorun(() => {
+    this.subscribe("transactionInfo", Meteor.userId());
+    const transactionDetail = Wallet.find().fetch();
+    this.state.set("details", transactionDetail[0]);
+  });
+});
+Template.payWithWallet.events({
+  "click #pay-with-wallet": (event) => {
+    event.preventDefault();
+    const balances = Template.instance().state.get("details").balance;
+    const cartAmounts = parseFloat(Cart.findOne({userId: Meteor.userId()}).cartTotal(), 10);
+    const pin = parseInt(document.getElementById("walletpin").value, 10);
+    const convertedPin = pin.toString();
+    const walletPin = Wallet.find().fetch()[0].userPin;
+    const comparePin = bcrypt.compareSync(convertedPin, walletPin);
+
+    if (comparePin) {
+      Alert({
+        title: "Are you sure you want to Purchase this product?",
+        text: "Funds will be deducted from your wallet",
+        type: "warning",
+        showConfirmButton: true,
+        cancelButtonText: "Dismiss",
+        showCancelButton: true
+      }).then(() => {
+        const cartAmount = cartAmounts;
+        const balance = balances;
+        if (cartAmount > balance) {
+          Alerts.toast("Insufficient balance", "error");
+          return false;
+        }
+        const currency = Shops.findOne().currency;
+        transactionId = Random.id();
+        Meteor.call("wallet/transaction", Meteor.userId(), {
+          amount: cartAmount,
+          date: new Date(),
+          orderId: transactionId,
+          transactionType: "Debit"
+        }, (err, res) => {
+          if (res) {
+            const paymentMethod = {
```

Comment:
Since the properties here doesn't depend on the function, you can as well move the variables outside to make the function lean.